### ISSUE TITLE
clang -Wshadow warning

### ISSUE THREAD

**Issue by mytbk** [MikePopoloski/slang#496]
```
/home/irudog/SourceCode/slang/source/compilation/builtins/MiscSystemFuncs.cpp:318:39: error: declaration shadows a structured binding [-Werror,-Wshadow]
                    [&visitor](auto&& arg) {
                                      ^
/home/irudog/SourceCode/slang/source/compilation/builtins/MiscSystemFuncs.cpp:316:33: note: previous declaration is here
            for (auto& [formal, arg] : aie.arguments) {
                                ^
1 error generated.
```

**Comment by Kuree** [comment]
I can confirm this with `clang-13`.

**Comment by MikePopoloski** [comment]
Fixed in 32458b8bba69d8928220ed9b3f0155018ee59d72
